### DISCUSSION TITLE
Invoice creation

### DISCUSSION THREAD

**Discussion by ofarrill** [kimai/kimai#3507] [General]
First of all, the obligatory thank you for the effort with kimai!

When creating invoices 

- I am not sure why the invoice templates need to be selected in the invoices filter dialog. Instead I believe the template should be related to a customer and/or a project. (For most customers I have to switch to a specific invoice template, because of different tax rules, required infos - like reverse charge, different banking accounts for different currencies, etc. So for this scenario "Save all" can't be used.)
- I think the "Mark as exported" should become default or better the checkbox should remember its latest state. (BTW, during invoice creation I did forget to mark the entries as exported. How can I mark them as exported without creating a new invoice?)

Thanks, best regards,

**Comment by kevinpapst** [comment]
- True, invoice templates can be linked to customers and changeable per invoice creation in the next major release. Wanna test? See #2902 
- It is default since a couple of releases. State remember => no. Export otherwise: batch update in timesheet listing or export screen.

**Comment by ofarrill** [comment]
Thanks Kevin, time to update here! Looking forward to https://github.com/kevinpapst/kimai2/pull/2902
Thanks, best regards,

**Comment by alexander-potemkin** [comment]
@kevinpapst , I would like to jump in here and ask you why "mark as exported" can't be fine tuned?

In my workflow (and I believe I'm not the only one here) I do the report first, then I preview it, quite often adjust something (like the template) and only then I do export; since this option is ticked every time I change anything on the page, I loose a perfect option that I had at Harvest - ability to quickly get all uninvoiced records.

The way it's done at Harvest, it's that the records are only marked as invoiced, when the invoice is saved (as opposed to previewed).

Is there any chance to get this logic at Kamai as well?

**Reply by kevinpapst** [reply]
The invoice preview never exports timesheets. So it is already the way you describe it...

**Reply by alexander-potemkin** [reply]
Hm, thank you! What about renaming 'Mark records as exported on invoice saving' to avoid any further possible confusion?

**Reply by kevinpapst** [reply]
You are the first one (that I remember) that found it easier to ask here than to test 😁

**Reply by alexander-potemkin** [reply]
I was the one who was just too careful not to mess with it, until now, when I find time and resources to submit it here; I would assume there are a lot of people like me, as it's money and you don't want to screw things in there.

It's definitely your call, but I'm the kind of person who likes to make his clients life easier, when possible.
